File: app/src/main/java/com/example/vintageradioapp/ui/VideoPlayerState.kt
package com.example.vintageradioapp.ui

import com.example.vintageradioapp.data.Song

data class VideoPlayerState(
    val songs: List<Song> = List(1) { Song() },
    val currentSongIndex: Int = 0,
    val isPlaying: Boolean = false,
    val currentPlaybackTimeSeconds: Int = 0,
    val totalDurationSeconds: Int = 0,
    val isLoading: Boolean = true,
    val error: String? = null,
    val triggerRestart: Boolean = false
) {
    val currentSong: Song?
        get() = songs.getOrNull(currentSongIndex)
}

File: VintageRadioApp/VintageRadioApp/ui/VideoPlayerState.swift
import Foundation

struct VideoPlayerState {
    var songs: [Song] = []
    var currentSongIndex: Int = 0
    var isPlaying: Bool = false
    var currentPlaybackTimeSeconds: Double = 0
    var totalDurationSeconds: Double = 0
    var isLoading: Bool = true
    var error: String? = nil
    var currentDecade: String = "70"
    var isTabletAndLandscape: Bool = false
    var isPrevButtonEnabled: Bool = false

    var currentSong: Song? {
        guard !songs.isEmpty, songs.indices.contains(currentSongIndex) else {
            return nil
        }
        return songs[currentSongIndex]
    }
}
